src/nprintml/label/aggregator/pcap.py

```diff
@@ -48,93 +55,113 @@ def normalize_npt(cls, npt_csv, path_input_base=None):
                 f"data result but stream was empty: {npt_csv}"
             )
 
+    @staticmethod
+    @storeresults
+    def flatten_npt(indexed_files):
+        print('Loading nPrints')
+
+        # We need every sample size to be the same, so we'll find the
+        # largest nPrint in all of the files and pad each sample with
+        # empty nPrints.
+        header = usecols = None
+        max_length = file_count = 0
+
+        for (file_count, (file_index, npt_file)) in enumerate(indexed_files, 1):
+            if header is None:
+                header = np.genfromtxt(npt_file, delimiter=',', max_rows=1, dtype=str)
+                usecols = np.arange(1, len(header))  # ignore data index (src_ip)
+                skip_header = 1 if isinstance(npt_file, (str, pathlib.Path)) else 0
+            else:
+                skip_header = 1
+
+            npt = np.genfromtxt(
+                npt_file,
+                delimiter=',',
+                skip_header=skip_header,
+                usecols=usecols,
+                dtype=NPT_DTYPE,
+            )
+
+            npt_dim1 = len(npt.shape) == 1
+            npt_length = 1 if npt_dim1 else len(npt)
+            max_length = max(max_length, npt_length)
+
+            npt_flat = npt if npt_dim1 else npt.ravel()
+
+            yield list(itertools.chain([file_index], npt_flat))
```

Comment:
As discussed elsewhere, this is how we then smuggle our row of index+data through into Pandas, (despite its heterogeneous typing).

It's entirely possible that this can be further optimized (either for performance or simplicity), but it seems to work pretty well as is.